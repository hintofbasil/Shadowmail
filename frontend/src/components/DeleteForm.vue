<template>
  <div>
    <div class="row">
      <div class="offset-by-two eight columns">
        <div class="warning1">
          Warning!
        </div>
      </div>
    </div>
    <div class="row">
      <div class="offset-by-two eight columns">
        <div class="warning2">
          This operation is not reversible.
        </div>
      </div>
    </div>
    <div class="row">
      <div class="offset-by-two eight columns">
        <form method="POST" id="click-me-form">
          <input
            type="email"
            name="email"
            id="click-me-text-input"
            placeholder="you@shadowmail.co.uk"
            v-model="email"
          />
          <br />
          <input
            type="submit"
            id="click-me-button"
            value="Confirm delete"
            v-on:click="submit"
          />
        </form>
      </div>
    </div>
    <DeleteFormResponse
      v-if="status"
      v-bind:status="status"
      v-bind:reason="reason"
    />
  </div>
</template>

<script>
import axios from "axios";
import DeleteFormResponse from "./DeleteFormResponse.vue";

const DELETE_URL = "/api/delete";

export default {
  components: {
    DeleteFormResponse
  },
  data: () => {
    const url = new URL(document.URL);
    const email = url.searchParams.get("email");
    const timestamp = url.searchParams.get("timestamp");
    const token = url.searchParams.get("token");

    return {
      email: email,
      timestamp: timestamp,
      token: token,
      status: "",
      reason: ""
    };
  },
  methods: {
    submit: async function(event) {
      event.preventDefault();
      const data = {
        email: this.email,
        timestamp: this.timestamp,
        token: this.token
      };
      try {
        const response = await axios.post(DELETE_URL, data);
        this.status = response.data.status;
        this.reason = "Email deleted";
      } catch (error) {
        const { response } = error;
        this.status = response.data.status || "ERROR";
        this.reason = response.data.reason || "An unexpected error occured";
      }
    }
  }
};
</script>

<style lang="scss" scoped>
@import "../colors";

.warning1,
.warning2 {
  text-align: center;
  color: $warning-text;
  font-weight: bold;
}

.warning1 {
  padding-top: 10%;
  font-size: 1.75em;
}

.warning2 {
  font-size: 1.5em;
}

#click-me-button {
  max-width: 90%;
  word-wrap: break-word;
  white-space: normal;
  height: 100%;
  margin-bottom: 0;
}

#click-me-form {
  text-align: center;
  padding-top: 10%;
  padding-bottom: 10%;
  margin-bottom: 0;
}

#click-me-text-input {
  text-align: center;
  margin-bottom: 5rem;
  width: 100%;
  font-size: 1.4em;
}

@media (min-width: 550px) {
  #click-me-form {
    padding-top: 6%;
    padding-bottom: 6%;
  }

  #click-me-button {
    font-size: 1em;
  }

  .request-delete-email {
    font-size: 2em;
    padding: 5%;
  }

  .warning1 {
    padding-top: 6%;
    font-size: 2.5em;
  }

  .warning2 {
    font-size: 2em;
  }
}
</style>
