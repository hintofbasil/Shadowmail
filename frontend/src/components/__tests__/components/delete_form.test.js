import React from 'react';
import renderer from 'react-test-renderer';

import DeleteForm from '../../delete_form';

describe('render', () => {
  it('renders successfully', () => {
    const tree = renderer
      .create(
        <DeleteForm />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
