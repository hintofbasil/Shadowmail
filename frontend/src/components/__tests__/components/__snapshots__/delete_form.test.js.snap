// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render renders successfully 1`] = `
Array [
  <div
    className="row"
  >
    <div
      className="offset-by-two eight columns"
    >
      <div
        className="warning1"
      >
        Warning!
      </div>
    </div>
  </div>,
  <div
    className="row"
  >
    <div
      className="offset-by-two eight columns"
    >
      <div
        className="warning2"
      >
        This operation is not reversible.
      </div>
    </div>
  </div>,
  <div
    className="row"
  >
    <div
      className="offset-by-two eight columns"
    >
      <form
        id="click-me-form"
        method="POST"
      >
        <input
          id="click-me-text-input"
          name="email"
          placeholder="you@domain.com"
          readOnly={true}
          type="email"
          value={null}
        />
        <br />
        <input
          id="click-me-button"
          onClick={[Function]}
          type="submit"
          value="Confirm delete"
        />
      </form>
    </div>
  </div>,
]
`;
